src/scripts.js

```diff
@@ -1,12 +1,129 @@
-var assert = require('assert')
-var ops = require('./opcodes')
-var typeForce = require('typeforce')
+var bufferutils = require('./bufferutils')
+var typeforce = require('typeforce')
+var types = require('./types')
 
+var ECSignature = require('./ecsignature')
 var ecurve = require('ecurve')
 var curve = ecurve.getCurveByName('secp256k1')
 
-var ECSignature = require('./ecsignature')
-var Script = require('./script')
+var OPS = require('./opcodes')
+var REVERSE_OPS = []
+for (var op in OPS) {
+  var code = OPS[op]
+  REVERSE_OPS[code] = op
+}
+
+function toASM (chunks) {
```

Comment:
For the to and from ASM function pair, perhaps they should be strict reverse of each other in terms of input and output types? Similarly to what has been suggested below, consider adding both the source type and destination types to the function name, e.g. `chunksArrayToASM`